hack/verify-spelling.sh

```diff
@@ -19,19 +19,15 @@ set -o nounset
 set -o pipefail
 
 export KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
+export GO111MODULE=on
+export GOPROXY="${GOPROXY:-https://proxy.golang.org}"
 
-# Install tools we need, but only from vendor/...
-cd ${KUBE_ROOT}
-go install ./vendor/github.com/client9/misspell/cmd/misspell
-if ! which misspell >/dev/null 2>&1; then
-    echo "Can't find misspell - is your GOPATH 'bin' in your PATH?" >&2
-    echo "  GOPATH: ${GOPATH}" >&2
-    echo "  PATH:   ${PATH}" >&2
-    exit 1
-fi
+# Pick out version of misspell from go.mod
```

Comment:
noting for the future: If we ever expect this repo to contain actual go tools for consumption (I'd guess not but) then we should consider making this something like `hack/tools/go.mod` and then cd there at the beginning of this script to do install, so we separate dev and compile time deps.

I'd guess we don't need to do this? BUT it also removes root-level repo clutter (`go.mod` and `go.sum` don't need to be in the root directory), and I'd suggest that this is a good pattern to spread throughout our repos. A `tools.go` to go with this is recommended upstream by the go team.

Reply:
Well there's also the `generator/` directory that has dependencies. The current go.mod covers both.

Reply:
right, you'd maybe use a distinct go.mod for generator (or the repo wide one, depending on if we only consider it a dev time tool for this repo or something meant for re-use).

something to think about in the future, absolutely not necessary for this PR.